components/CanarySearch.tsx

```diff
@@ -0,0 +1,62 @@
+import React from "react";
+
+export default function CanarySearch() {
+  const [loaded, setLoaded] = React.useState(false);
+
+  const TABS = JSON.stringify([
+    {
+      name: "Docs",
+      pattern: "**/langfuse.com/**",
+    },
+    {
+      name: "GitHub",
+      pattern: "**/github.com/**",
+    },
+  ]);
+
+  React.useEffect(() => {
+    Promise.all([
+      import("@getcanary/web/components/canary-root.js"),
+      import("@getcanary/web/components/canary-provider-cloud.js"),
+      import("@getcanary/web/components/canary-modal.js"),
+      import("@getcanary/web/components/canary-trigger-searchbar.js"),
+      import("@getcanary/web/components/canary-content.js"),
+      import("@getcanary/web/components/canary-input.js"),
+      import("@getcanary/web/components/canary-search.js"),
+      import("@getcanary/web/components/canary-search-results.js"),
+      import("@getcanary/web/components/canary-search-match-github-issue.js"),
+      import(
+        "@getcanary/web/components/canary-search-match-github-discussion.js"
+      ),
+      import("@getcanary/web/components/canary-filter-tabs-glob.js"),
+      import("@getcanary/web/components/canary-footer.js"),
+    ]).then(() => {
+      setLoaded(true);
+    });
+  }, []);
+
+  if (!loaded) {
+    return null;
+  }
+
+  return (
+    <canary-root framework="nextra">
+      <canary-provider-cloud project-key="cpa0aae64b">
```

Comment:
Based on the canary [docs](https://getcanary.dev/docs/local/integrations/nextra.html), canary is not available without canary cloud when using nextra and you added a key here. Do you know whether a local options exists as keyword search would be nice already across github and docs?

Reply:
For Nextra, hooking into the build step and configuring a custom local search index is a bit more involved, but it is doable.

If you really want to avoid hosted search and leverage existing sources like https://github.com/langfuse/langfuse-docs/blob/main/src/langfuse_github_discussions.json, I can write a small script that hooks into the webpack build step to index statically generated HTML and the `langfuse_github_discussions.json` file. You can then search it using `canary-provider-cloud`.

However, 1 - I need issue #865 to be resolved, and 2 - it might take some time.

Just FYI, Canary is self-hostable. You can start using it now and move to self-hosting later.

Feel free to clarify if you're only interested in local search at this time, which is perfectly fine.

Reply:
thanks for the clarification!

Is there a way for maintainers to use/access/configure the hosted search? I cannot find public pricing or any information on canary cloud

Self-hosting this is currently a bad use of our time/effort. I'd rather keep the current search, or go for something new like canary that is either local or managed-cloud

Reply:
@marcklingen 

We have pricing page here: https://getcanary.dev/docs/cloud/platform/pricing
Should take `<3min` to setup. https://getcanary.dev/docs/cloud/platform/manual

`Canary Cloud` is at https://cloud.getcanary.dev